R/integrate_sces.R

```diff
@@ -0,0 +1,193 @@
+# This function can perform the crux integration based on the given integration method
+
+
+#' Integrate a combined set of SingleCellExperiment objects using a specified
+#'  integration method. The final SCE object will contain an additional reducedDim
+#'  entitled `{integration_method}_PCA`.
+#'
+#' @param combined_sce A combined SCE object as prepared by `scpcaTools::merge_sce_list()`.
+#' @param integration_method The integration method to use. One of `fastMNN` or
+#'  `harmony` (case-insensitive)
+#' @param batch_column The column in the combined SCE object indicating batches
+#'  being integrated. Default is "sample".
+#' @param retain_assays A character vector of uncorrected assays to retain in
+#'   the integrated SCE object. By default, both "counts" and "logcounts" are retained.
+#' @param return_corrected_expression A boolean indicating whether corrected expression
+#'   values determined by the given integration method should be included in the
+#'   integrated SCE object. Note that `harmony` does not calculate corrected expression,
+#'   so this argument is ignored for this integration method. Default is `TRUE`.
+#' @param harmony_do_PCA A `harmony`-specific argument (ignored otherwise) indicating whether
+#'   integration should perform PCA. If `FALSE` (default), `harmony` will use the existing PCs to perform
+#'   integration. If `TRUE`, `harmony` will will start from normalized expression when integrating and
+#'   re-calculate PCs.
```

Comment:
I do not think we will ever start with corrected expression, no, so this can be removed entirely. We can explain in docs accordingly.
This is what `harmony` will do if you ask it to calculate PCs, just fwiw - 

https://github.com/immunogenomics/harmony/blob/63ebd73199b79ae0d7563425a765d76deb916fe3/R/ui.R#L125-L129